src/util/api_quote_updater.ts

```diff
@@ -0,0 +1,162 @@
+import { BigNumber } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import { Dispatch } from 'redux';
+import { ZRX_API_URL } from '../constants';
+
+import { Action, actions } from '../redux/actions';
+import { QuoteFetchOrigin, SwapQuoteResponse, TokenInfo } from '../types';
+
+import { analytics } from './analytics';
+
+
+import { errorFlasher } from './error_flasher';
+import { errorReporter } from './error_reporter';
+import { tokenUtils } from './token';
+
+
+export const apiQuoteUpdater = {
+    updateSwapQuoteAsync: async (
+        dispatch: Dispatch<Action>,
+        isIn: boolean,
+        takerAddress: String,
+        tokenIn: TokenInfo,
+        tokenOut: TokenInfo,
+        tokenUnitAmount: BigNumber,
+        fetchOrigin: QuoteFetchOrigin,
+        options: {
+            setPending: boolean;
+            dispatchErrors: boolean;
+        },
+        skipValidation: boolean = true,
+    ): Promise<void> => {
+        const tokenSell = isIn ? tokenIn : tokenOut;
+        const tokenBuy = isIn ? tokenOut : tokenIn;
+
+
+        const tokenAddressSell = tokenSell.address;
+        const tokenAddressBuy = tokenBuy.address;
+
+        if (options.setPending) {
+            // mark quote as pending
+            dispatch(actions.setQuoteRequestStatePending());
+        }
+
+        let newSwapQuote: SwapQuoteResponse | undefined;
+        try {
+            let takerAddressString = '';
+            let skipValidationString = '';
+            if (takerAddress) {
+                takerAddressString = `&takerAddress=${takerAddress}`
+            }
+            if (skipValidation) {
+                skipValidationString = `&skipValidation=true`
+            }
+
+            const response = await fetch(`${ZRX_API_URL}/quote?sellToken=${tokenAddressSell}&buyToken=${tokenAddressBuy}&sellAmount=${tokenUnitAmount}${takerAddressString}${skipValidationString}`);
+            if (response.ok && response.status === 200) {
```

Comment:
NIT: no need to check twice :)
```suggestion
            if (response.status === 200) {
```